manifests/redhat_installer.pp

```diff
@@ -0,0 +1,162 @@
+#
+# @param api_key String:Your DataDog API Key. Please replace with your key value.
+# @param datadog_site String: The site of the Datadog intake to send Agent data to. Defaults to 'datadoghq.com'.
+# @param agent_major_version Integer: The major version of the Datadog agent to install. Defaults to 7.
+# @param agent_minor_version Optional[String]: The minor version of the Datadog agent to install.
+# @param installer_repo_uri Optional[String]: The URI of the installer repository.
+# @param rpm_repo_gpgcheck Optional[Boolean]: Whether to check the GPG signature of the repository.
+# @param apm_instrumentation_enabled Optional[Enum['host', 'docker', 'all']]: Enable APM instrumentation for the specified environment (host, docker, or all).
+# @param apm_instrumentation_libraries_str Optional[String]: APM instrumentation libraries as a comma-separated string.
+# @param remote_updates Boolean: Whether to enable Agent remote updates. Default: false.
+#
+class datadog_agent::redhat_installer (
+  String $api_key = 'your_API_key',
+  String $datadog_site = $datadog_agent::params::datadog_site,
+  Integer $agent_major_version = $datadog_agent::params::default_agent_major_version,
+  Optional[String] $agent_minor_version = undef,
+  Optional[String] $installer_repo_uri = undef,
+  Optional[Boolean] $rpm_repo_gpgcheck = undef,
+  Optional[Enum['host', 'docker', 'all']] $apm_instrumentation_enabled = undef,
+  Optional[String] $apm_instrumentation_libraries_str = undef,
+  Boolean $remote_updates = $datadog_agent::params::remote_updates,
+) inherits datadog_agent::params {
+  # Generate installer trace ID as a random 64-bit integer (Puppet does not support 128-bit integers)
+  # Note: we cannot use fqdn_rand as the seed is dependent on the node, meaning the same trace ID would be generated on each run (for the same node)
+  # -An: no address, no leading 0s
+  # -N8: read 8 bytes
+  # -tu8: unsigned integer, 8 bytes (64 bits)
+  exec { 'Generate trace ID':
+    command => "echo $(od -An -N8 -tu8 < /dev/urandom | tr -d ' ') > /tmp/datadog_trace_id",
+    path    => ['/usr/bin', '/bin'],
+    onlyif  => '/bin/sh -c "command -v tr && command -v od && command -v echo"',
+  }
+
+  # Start timer (note: Puppet is not able to measure time directly as it's against its paradigm)
+  exec { 'Start timer':
+    command => 'date +%s%N > /tmp/puppet_start_time',
+    path    => ['/usr/bin', '/bin'],
+    require => Exec['Generate trace ID'],
+  }
+
+  # Define the GPG keys to use for the repository
+  # We only use the latest key and previous key since the installer is signed with the latest key and the agent might be signed with the previous key.
+  $all_keys = [
+    'https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public',
+    # Previous, EOL September 2024
+    'https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public',
+    # Current
+    'https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public',
+  ]
```

Comment:
When using GPG keys we need to also import the future key to prevent customers to have to do any key rotation related work on the next rotation following their installation.
```suggestion
  $all_keys = [
    'https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public',
    # Previous, EOL September 2024
    'https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public',
    # Current
    'https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public',
    # Future, active from April 2028
    'https://keys.datadoghq.com/DATADOG_RPM_KEY_4F09D16B.public',
  ]
```

Reply:
Adressed by 9be1b567f0a748a28ad4d1b583f5960489f27a37